Suit Storage Unit changes

Chiquiii opened the pull request:
:cl: Orelbon 
bugfix: Suit storage units now check for access.
tweak: Suit storage units don't drop items when the powers fails anymore. 
tweak: Suit storage units can now be pried open when they are powered off and not locked.
tweak: Suit storage units can now dispense items when they are offline. 
/:cl:

- Now checks for access 
- No longer drops items when they lose power
- Can be crowbared
- Updates legacy shitcode 

Borrowed sprites from TG and modified their `update_icon()` code to fit ours.

Fixes #16856

HeyBanditoz commented:
I don't think it should check for access.


On 1/13/2018 6:46 PM, orelbon wrote:
>
>   * Now checks for access
>   * Picks items dropped after power is restored
>
> I'm getting an error when trying to get back the items that the unit 
> drops on the ground. Some help would be appreciated.
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/Baystation12/Baystation12/pull/20267
>
>
>         Commit Summary
>
>   * Suit Storage Unit
>
>
>         File Changes
>
>   * *M* code/game/machinery/suit_storage_unit.dm
>     <https://github.com/Baystation12/Baystation12/pull/20267/files#diff-0>
>     (32)
>
>
>         Patch Links:
>
>   * https://github.com/Baystation12/Baystation12/pull/20267.patch
>   * https://github.com/Baystation12/Baystation12/pull/20267.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Baystation12/Baystation12/pull/20267>, or mute the 
> thread 
> <https://github.com/notifications/unsubscribe-auth/AHOUiHypoT8idb6pmkdOYN_mrv841sWfks5tKVyQgaJpZM4RdcMZ>.
>

Chiquiii commented:
It does, different units have different access requirements.

Techhead0 commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -368,6 +372,7 @@
 		src.safetieson = !src.safetieson
 
 #define dispense_clothing(item) if(src.item){item.dropInto(loc); src.item = null}
+#define capture_clothing(item) if(!src.item){loc.item.dropInto(src); src.item = TRUE}
```
I try to be helpful, but here... [this feels like the only thing I can say](https://i.imgflip.com/21aqbc.jpg).

Orelbon replied:
plz send halp

comma commented:
I think it can be majorly ungrossified if instead of keeping a variable for each item, you'd just have a list with slots as keys, and used that. Can do just a single proc then.

PsiOmegaDelta commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -272,270 +265,228 @@
+
+		occupant.client.eye = occupant.client.mob
+		occupant.client.perspective = MOB_PERSPECTIVE
+	occupant.loc = loc
```
occupant.dropInto(loc)

Sbotkin commented:
>I don't think it should check for access.

What's the point of the locking mechanism then?

Chiquiii committed:
Suit Storage Unit 

- Now checks for access 
- No longer drops items when they lose power
Tried to improve old shit code


More shitcode clean up


Adds a new feedback define


More deshitting of code


Fix errors?


Final Things.

PsiOmegaDelta commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -272,270 +265,229 @@
 	if(..())
 		return
-	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
+	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
```
This is all handled by `..()`. Can be removed.

PsiOmegaDelta commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -272,270 +265,229 @@
+	if(stat & NOPOWER)
+		to_chat(user, "<span class='warning'>The unit is offline.</span>")
+		return
+	if(!allowed(usr))
```
Use `user`, not `usr`

PsiOmegaDelta commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -272,270 +265,229 @@
+		to_chat(user, "<span class='warning'>The unit is offline.</span>")
+		return
+	if(!allowed(usr))
+		to_chat(usr, FEEDBACK_ACCESS_DENIED)
```
As above.

PsiOmegaDelta commented:
I suppose suit storage code isn't worse than it was before.
